src/Zio/UPath.cs

```diff
@@ -296,165 +277,159 @@ internal static StringBuilder GetSharedStringBuilder()
         var parts = internalHelper.Slices;
         parts.Clear();
 
-        var builder = internalHelper.Builder;
```

Comment:
I moved the builder to below where it's actually used, so it's not needlessly set when not used.